File: GARI_Umbria_2026-08-03/DotNet/AgronicaCoreStd/AgronicaCoreModelsSTD/anagrafiche/ParticelleCatastali.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace AgronicaCoreModelsSTD.anagrafiche
{
    public class ParticelleCatastali
    {

        public PK primaryKey { get; set; }

        public double Area { get; set; }

        public List<ParticelleCatastaliMacrouso> macrousi { get; set; }

        public List<ParticelleCatastaliZona> zonizzazione { get; set; }

        public List<ParticelleCatastaliClassamento> classamento { get; set; }

        public List<ParticelleCatastaliMetodoProduzione> metodoProduzione { get; set; }

        public string proprietario { get; set; }

        public ParticelleCatastali(PK primaryKey)
        {
            this.primaryKey = primaryKey;
        }

        public ParticelleCatastali()
        {

        }

        public class PK
        {

            public string Prov { get; set; }
            public string Com { get; set; }
            public string Sezione { get; set; }
            public int Foglio { get; set; }
            public int Numero { get; set; }
            public string Subalterno { get; set; }

            public PK(string Prov, string Com, string Sezione, int Foglio, int Numero, string Subalterno)
            {
                this.Prov = Prov;
                this.Com = Com;
                this.Sezione = Sezione;
                this.Foglio = Foglio;
                this.Numero = Numero;
                this.Subalterno = Subalterno;
            }

            public PK()
            {

            }
        }

    }
}

File: GARI_Umbria_2026-08-03/GiasNG/src/app/Model/anagrafiche/ParticelleCatastali.ts
import { ParticelleCatastaliClassamento } from './ParticelleCatastaliClassamento';
import { ParticelleCatastaliMacrouso } from './ParticelleCatastaliMacrouso';
import { ParticelleCatastaliMetodoProduzione } from './ParticelleCatastaliMetodoProduzione';
import { ParticelleCatastaliZona } from './ParticelleCatastaliZona';

export type PKParticelleCatastali = typeof ParticelleCatastali.PK.prototype;
export class ParticelleCatastali {

  Area: number;
  macrousi: ParticelleCatastaliMacrouso[];
  zonizzazione: ParticelleCatastaliZona[];
  classamento: ParticelleCatastaliClassamento[];
  metodoProduzione: ParticelleCatastaliMetodoProduzione[];
  proprietario: string;

  primaryKey: PKParticelleCatastali;

  public static PK = class PKParticelleCatastali {
    Prov: string;
    Com: string;
    Sezione: string;
    Foglio: number;
    Numero: number;
    Subalterno: string;

    constructor (
      Prov: string,
      Com: string,
      Sezione: string,
      Foglio: number,
      Numero: number,
      Subalterno: string
    ) {
      this.Prov = Prov;
      this.Com = Com;
      this.Sezione = Sezione;
      this.Foglio = Foglio;
      this.Numero = Numero;
      this.Subalterno = Subalterno;
    }
  };
}
